### PR TITLE
Add process hooks to tell systemd our state.

### PR DESCRIPTION
**Pull request by erikjohnston**
Adds new dependency on `sdnotify`, but `python3-sdnotify` packages exists on debian stretch so I think that is fine.

### CODE REVIEW

**Comment by erikjohnston** [comment]
Is there anywhere suitable this needs to be doc'ed?

File: synapse/app/_base.py [hunk 1]
```diff
@@ -292,6 +302,25 @@ def setup_sentry(hs):
...
+    """
+
+    # Tell systemd our state, if we're using it. This will silently fail if
+    # we're not using systemd.
+    sd_channel = sdnotify.SystemdNotifier()
+
+    hs.get_reactor().addSystemEventTrigger(
+        "after",
+        "startup",
+        lambda: sd_channel.notify("READY=1\nMAINPID=%s" % (os.getpid())),
```
Comment (richvdh) [inline synapse/app/_base.py]: doesn't this need to happen a bit later? IIRC we do some acme foo after the reactor starts but before we bind the http and replication listeners

Comment (erikjohnston) [reply synapse/app/_base.py]: Oh, it looked like we started listening pretty much immediately we start the reactor, but actually we do some acme stuff first. Good spot.

File: contrib/systemd/matrix-synapse.service [hunk 2]
```diff
@@ -14,7 +14,9 @@
 Description=Synapse Matrix homeserver
 
 [Service]
-Type=simple
+Type=notify
```
Comment (richvdh) [inline contrib/systemd/matrix-synapse.service]: there's also systemd-with-workers, fwiw

**Pull request merged**